opossum-tool/OpossumUI
# Introduce QA mode

### Summary of changes

This PR adds a new QA mode. The user should only be able to mark attributions as preferred ( or unmark if preferred beforehand) if this mode is enabled. The mode can be enabled by clicking the respective item in the `View` submenu.

### Context and reason for change

We want to restrict the feature to mark attributions as preferred.

### How can the changes be tested

Besides the unit and integration tests:
Check out the commit and open opossum_input.json from the example-files. Click on the index.tsx file in the resource tree and see one attribution with a grey star. If you open the context menu ... for this attribution you only see the options Mark for replacement and Delete. Enable QA mode by clicking the menu item below View and observe that there is an additional option Mark as preferred. If you prefer this attribution and save it, the option will change to Unmark as preferred. If you once again change the mode by clicking the respective menu item, the option will disappear.
